react/week3/todo-week3/src/todo-app.js

```diff
@@ -0,0 +1,210 @@
+  }
+
+  function onEditTime(event) {
+    const value = event.target.value
+    editTime(id, value)
+  }
+
+  return (
+    <div>
+      <div style={{ textDecoration: checked ? "line-through" : "none" }}>
+        {Editing ? (
+          <input type="text" value={description} onChange={editing} />
+        ) : (
+          <h3>{description}</h3>
+        )}
+        {Editing ? (
+          <input
+            type="date"
+            id="deadline"
+            name="deadline"
+            value={deadline}
+            onChange={onEditTime}
+          />
+        ) : (
+          <h4>{deadline}</h4>
+        )}
+      </div>
+      <input type="checkbox" checked={checked} onChange={Check} />
+      <button onClick={Delete}>Delete</button>
+      <button onClick={() => setEditing(!Editing)}>
+        {Editing ? "Update" : "Edit"}
+      </button>
+    </div>
+  )
+}
+
+function AddTodoForm({ addTodo }) {
+  const [description, setDescription] = useState("")
+  const [deadline, setDeadline] = useState("")
+
+  function onSubmit(event) {
+    event.preventDefault()
+
+    addTodo(description, deadline)
+
+    setDescription("")
+    setDeadline("")
+  }
+
+  const editing = event => {
+    const value = event.target.value
+    console.log(value)
+    setDescription(value)
+  }
+
+  const onEditTime = event => {
+    const value = event.target.value
+    setDeadline(value)
+  }
+
+  return (
+    <form onSubmit={onSubmit}>
+      <div>
+        <label htmlFor="description">Description</label>
+        <input
+          type="text"
+          name="description"
+          id="description"
+          value={description}
+          onChange={editing}
+        />
+      </div>
+      <div>
+        <label htmlFor="deadline"></label>
+        <input
+          type="date"
+          id="deadline"
+          name="deadline"
+          value={deadline}
+          onChange={onEditTime}
+        />
+      </div>
+      <button type="submit">Add Todo</button>
+    </form>
+  )
+}
+
+function Todos() {
+  const [todos, setTodos] = useState([])
+
+  useEffect(() => {
+    fetch(URL)
+      .then(res => res.json())
+      .then(data => {
+        setTodos(data)
+      })
+  }, [])
+
+  function checking(id) {
+    setTodos(prevTodos => {
+      return prevTodos.map(todo => {
+        if (todo.id === id) {
+          return { ...todo, checked: !todo.checked }
+        } else {
+          return todo
+        }
+      })
+    })
+  }
+
+  function addTodo(description, deadline) {
+    //   if any input is empty
+    if (!description || !deadline) {
+      alert("Description and deadline should not be empty")
+      return
+    }
+
+    // if dealine is before current date
+    const inputDate = Date.parse(deadline)
+    const currentDate = Date.now()
+    if (inputDate < currentDate) {
+      alert("date should not be before current date")
+      return
+    }
+
+    setTodos(prevTodos => {
+      const newList = {
+        id: prevTodos.length + 1,
```

Comment:
This will give you problems when someone deletes a todo, because you can end up with equal ids. 
Say you have the following todos:
```
const todos = [
  {id: 1},
  {id: 2},
  {id: 3}
] 
```
and the user deletes the second todo.
When they add a todo next, it'll get the id "3". (Length of list + 1)